src/extensions.rs

```diff
@@ -195,4 +195,15 @@ impl<T: AsRef<str>> T {
         let index = s.find(c)?;
         Some((&s[..index], &s[index + c.len_utf8()..]))
     }
+
+    fn split_at_word<'a>(&'a self, split_at: &str) -> (String, String) {
+        let mut words = self.as_ref().trim().split(' ').collect_vec();
+        match words.iter().position(|w| w == &split_at) {
+            Some(word_ind) => {
+                let right_side = words.split_off(word_ind + 1).join(" ");
+                return (words.join(" "), right_side);
+            }
+            None => return (String::from(self.as_ref()), String::new()),
```

Comment:
No need for returns here, as rust will just treat the last expression in any function (as long as it doesn't have a semicolon) as the return value. This means that here this is equivalent to doing `return match words.iter().etcetcetc` - which then again is equivalent to just doing the match without the `return` keyword, as it's implicitly getting returned bcos it's the last expression in the function, and doesn't have a semicolon.
```suggestion
            Some(word_ind) => {
                let right_side = words.split_off(word_ind + 1).join(" ");
                (words.join(" "), right_side)
            }
            None => (String::from(self.as_ref()), String::new()),
```